### DISCUSSION TITLE
[x] [Topic 3 Day 2] Shaping a bioinformatics software accelerator program

### DISCUSSION THREAD

**Discussion by supernord** [RSEAA/RSEAA.github.io#51] [Discussion topics for #RSEAA2023]
## Topic

RSEs should be supported to create high quality reusable software that is performant on high end compute systems. 

A potential way to address this is a software accelerator program that supports development, optimisation and sharing: one that provides software authors with the expertise and resources required to follow best practice on compute systems in Australia and internationally.

The software accelerator vision aims to create a culture of creating, optimising and sharing best practice bioinformatics on peak infrastructures, with an emphasis on building performant and scaled-up tools and workflows.

> **We would like to collectively discuss and shape this idea together with the RSE community.**

## Background

The accelerator idea arose from an existing program of work called the Australian BioCommons Leadership Share (ABLeS). An ABLeS accelerator project is envisioned to provide environments on peak compute systems that allow bioinformatics focused RSEs to test, scale, debug and enhance software (including tools and workflows), and pair this with support and expertise in making software FAIR. This concept is being actively pursued as a collaborative effort by the Australian BioCommons, Pawsey Supercomputing Centre, New Zealand eScience Infrastructure and the National Computational Infrastructure.

**Comment by manodeep** [comment]
GIven the unconference theme "Silos to Synergy - Achieving collaboration across domains", maybe we can also discuss how to share the expertise across domains?

**Reply by supernord** [reply]
Thanks @manodeep 😄 
And absolutely agree. It would be ideal to share learnings from the software accelerator across domains, and with more of our colleagues internationally.

**Comment by pr4deepr** [comment]
Agree with @manodeep , for example, in the bio image analysis world, the use of high performance compute and workflow management systems are not as common.
It would be great to learn from bioinformatics where the systems are quite mature.

**Comment by SaranjeetKaur** [comment]
How can the international RSE community support this programme? Who would be the point of contact?

**Comment by smanosmxm** [comment]
**DISCUSSION NOTES Pt 1.**

- There's an inherent tension in getting the time to improve the software. 
- When there's long term funding, then you can put RSE effort into improvement and scaling.
- How do you support the shorter term, earlier-in-cycle, reactive stage of a software project?
- Is there value to have a dedicated RSE to provide hands on support for the top end of the problem, the specialised support required. But up to that point, it's guided and self supported.
- On the contrary, how do we guide people in the earlier stages, e.g. the progression to command line from GUI. 
- Upskilling to HPC is itself a grand challenge - and in many ways solutions (training etc.) exist for this. 
- Can't scale to all the potential RSE's who might have a use for HPC.

**Reply by smanosmxm** [reply]
Spatial multi-omics, single cell, and so on and so on - new multi-scale multi-modal data. These are related and overlapping areas that would be support through such a scheme.

**Reply by smanosmxm** [reply]
'Accelerator program' -> 'Shark Tank for RSE's' analogy. Getting RSE's to sit together and work together to solve interrelated challenges.

**Comment by SaranjeetKaur** [comment]
Rowland Mosbergen to Everyone (14 Sep 2023, 10:03)
Would you see it as an idea of helping improve their NCMAS application (in terms of showing how you have optimised for HPC)?

**Comment by SaranjeetKaur** [comment]
Ziad Al-Bkhetan to Everyone (14 Sep 2023, 10:04)
https://australianbiocommons.github.io/ables/

**Comment by pr4deepr** [comment]
Link to GESTALT (Global Alliance for Spatial Technologies (GESTALT) ):

https://twitter.com/LGMartelotto/status/1688380146548703232

**Reply by pr4deepr** [reply]
they have a twitter account:
https://twitter.com/GESTALT_sp

**Reply by pr4deepr** [reply]
List of reasons why this was started:
https://twitter.com/GESTALT_sp/status/1689877573323624448
